compiler/luci/service/src/Nodes/CircleBatchMatMul.cpp

```diff
@@ -30,4 +52,56 @@ luci::CircleNode *CloneNodeLet<CN::ABC>::visit(const luci::CircleBatchMatMul *no
   return cloned;
 }
 
+// BatchMatMulV2 supports broadcasting in the batch dimensions(BatchMatMul doesn't)
+// TODO Distinguish BatchMatMul and BatchMatMulV2
+loco::TensorShape sinf::Algorithm::visit(const luci::CircleBatchMatMul *node)
+{
+  const auto x = loco::must_cast<CircleNode *>(node->x());
+  const auto y = loco::must_cast<CircleNode *>(node->y());
+
+  const auto x_shape = sinf::circle_shape(x);
+  const auto y_shape = sinf::circle_shape(y);
+
+  uint32_t x_rank = x_shape.rank();
+  uint32_t y_rank = y_shape.rank();
+  assert(x_rank >= 2 && y_rank >= 2);
+
+  uint32_t max_rank = x_rank > y_rank ? x_rank : y_rank;
+  loco::TensorShape output_shape;
+  output_shape.rank(max_rank);
+
+  // broadcast in the batch dimensions
+  if (x_rank > 2 || y_rank > 2)
```

Comment:
if x_rank = 3, y_rank = 2, the condition also works,
In this case, y_batch_dims rank is 0, is this also allowed?
i want to know how it works because there is no test case.

Reply:
> In this case, y_batch_dims rank is 0, is this also allowed? 

Yes.  It is allowed. 

> i want to know how it works because there is no test case. 

I make an example for the case. 
<sub> I intentionally use ? instead of 0, because 0 means unknown in shape, here </sub> 
```
x: (?, 4, 3) ----  BatchMatMul --- out: (?, 4, 5) 
		  / 	
y: (3, 5)    ----/ 

```

Reply:
Ah, I a bit misunderstood your questions :smiling_face_with_tear: 

I put another working example of 2D - 3D batch matmul.  
```
x: (2, 4, 3) ----  BatchMatMul --- out: (2, 4, 5) 
		          / 	
y: (3, 5)    ----/ 
```